### PR TITLE
Add temporal models to model gallery

### PR DESCRIPTION
**Pull request by Jaleleddine**
**Description**

This pull request ... introduces plot functions for temporal model galeries

**Dear reviewer**

### CODE REVIEW

File: examples/models/temporal/plot_gaussian_temporal.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+r"""
+.. _gaussian-temporal-model:
+
+Gaussian Temporal Model
+=======================
+
+This model parametrises a gaussian time model.
+
```
Comment (adonath) [inline examples/models/temporal/plot_gaussian_temporal.py:8]: Model formula?

File: gammapy/modeling/models/temporal.py [hunk 2]
```diff
@@ -50,6 +50,28 @@ def time_sum(t_min, t_max):
...
+        """
+        Plot Temporal Model.
+
+        Parameters
+        ----------
+        time_range : `~astropy.time.Time`
+            times to plot the model
+        ax : `~matplotlib.axes.Axes`, optional
+            axis
+        """
```
Comment (adonath) [inline gammapy/modeling/models/temporal.py:63]: Please also document the return type like:
```
Returns
--------
ax : ....
```

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks a lot @Jaleleddine! I have left a few minor comments so far. 

It would be good to add one unit test for `TemporalModel.plot()` in https://github.com/gammapy/gammapy/blob/master/gammapy/modeling/models/tests/test_temporal.py. The pattern we use for this is the following:

```python3
@requires_dependency("matplotlib")
def test_plot_spectral_model():
    with mpl_plot_check():
        model.plot()
```

Where the `mpl_plot_check()` context is imported from `gammapy.utils.testing`

File: examples/models/temporal/plot_gaussian_temporal.py (hunk 1 above)
Comment (Jaleleddine) [reply examples/models/temporal/plot_gaussian_temporal.py:8]: done

File: examples/models/temporal/plot_constant_temporal.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+r"""
+.. _constant-temporal-model:
+
+Constant Temporal Model
+=======================
+
+This model parametrises a constant time model.
+
```
[the remark this answers is not in the document]
Comment (Jaleleddine) [reply examples/models/temporal/plot_constant_temporal.py:8]: done

File: examples/models/temporal/plot_gaussian_temporal.py (hunk 1 above)
Comment (Jaleleddine) [reply examples/models/temporal/plot_gaussian_temporal.py:8]: done

File: gammapy/modeling/models/temporal.py (hunk 2 above)
Comment (Jaleleddine) [reply gammapy/modeling/models/temporal.py:68]: done

File: examples/models/temporal/plot_expdecay_temporal.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
+r"""
+.. _constant-temporal-model:
+
+ExpDecay Temporal Model
+=======================
+
+This model parametrises an ExpDecay time model.
+
```
[the remark this answers is not in the document]
Comment (Jaleleddine) [reply examples/models/temporal/plot_expdecay_temporal.py:8]: done

**Review by adonath — APPROVED** [review]
Thanks @Jaleleddine! I have no further comments...

**Comment by adonath** [comment]
Thanks again @Jaleleddine, here is the rendered version:  https://docs.gammapy.org/dev/modeling/gallery/index.html#temporal-models

**Pull request merged**